pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -191,6 +191,27 @@ func validateMinMaxScale(config *autoscalerconfig.Config, m map[string]string) *
 		errs = errs.Also(validateMaxScaleWithinLimit(k, max, config.MaxScaleLimit))
 	}
 
+	// if MinNonZeroReplicas is also set, validate that min <= nz min <= max
+	if k, v, ok := MinNonZeroRepicas.Get(m); ok {
+		if nzMin, err := strconv.ParseInt(v, 10, 32); err != nil {
+			errs = errs.Also(apis.ErrInvalidValue(v, k))
+		} else if min > int32(nzMin) {
+			errs = errs.Also(&apis.FieldError{
+				Message: fmt.Sprintf("min-scale=%d is greater than min-non-zero-replicas=%d", min, nzMin),
+				Paths:   []string{MinNonZeroReplicasKey},
```

Comment:
nit: this probably should be `k` since we want to return the annotation key used. It's a noop for this annotation though